File: backend/src/main/scala/order/object/after-sales/ResolveAfterSalesRequest.scala
package domain.order

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class ResolveAfterSalesRequest(
    approved: ApprovalFlag,
    resolutionNote: ResolutionText,
    resolutionMode: Option[AfterSalesResolutionMode],
    actualCompensationCents: Option[CurrencyCents],
    couponMinimumSpendCents: Option[CurrencyCents],
)
object ResolveAfterSalesRequest:
  given Encoder[ResolveAfterSalesRequest] = deriveEncoder
  given Decoder[ResolveAfterSalesRequest] = deriveDecoder

File: frontend/src/order/object/after-sales/ResolveAfterSalesRequest.ts
import type {
  AfterSalesResolutionMode,
  ApprovalFlag,
  CurrencyCents,
  ResolutionText,
} from '@/shared/object/domain/DomainObjects'

export type ResolveAfterSalesRequest = {
  approved: ApprovalFlag
  resolutionNote: ResolutionText
  resolutionMode?: AfterSalesResolutionMode
  actualCompensationCents?: CurrencyCents
  couponMinimumSpendCents?: CurrencyCents
}
